### PR TITLE
An/iter5

### PR DESCRIPTION
**Pull request by AashnaNarang**

### CODE REVIEW

**Review by krishane2018 — APPROVED** [review]
I don't think I see any issues with the main code, just ensure logic is correct.

File: src/main/Elevator.java [hunk 1]
```diff
@@ -41,6 +41,7 @@
 	private ElevatorTimer elevatorTimer;
 	private boolean isDoorsOpen;
 	private boolean didTimeout;
+	private int error;
```
Comment (abdol99) [inline src/main/Elevator.java]: Would the name errorCode be better?

File: src/main/Elevator.java [hunk 2]
```diff
@@ -211,12 +213,16 @@ public void run() {
...
 			while (this.getIsDoorsOpen()) {};
 			if (this.getDidTimeout()) {
+				this.error = fe.getErrorCode();
+				System.out.println(Thread.currentThread().getName() + " is in error " + this.error + " state");
 				closeDoorsTask.cancel();
 				timer.cancel();
 				this.setDidTimeout(false);
 				fe.setErrorCode(0);
 			} else {
+				this.error = 0;
```
Comment (abdol99) [inline src/main/Elevator.java]: should we just set it to fe.getErrorCode() here?

File: src/main/Elevator.java (hunk 1 above)
Comment (AashnaNarang) [reply src/main/Elevator.java]: fixed

File: src/main/Elevator.java (hunk 2 above)
Comment (AashnaNarang) [reply src/main/Elevator.java]: fixed

**Pull request merged**